packages/headless/src/user/provider.tsx

```diff
@@ -32,6 +44,17 @@ export const UserProvider: React.FC<{ children: React.ReactNode }> = ({
   const { token, setBearerToken, removeToken } = useServer();
   const query = useCurrentUser(!!token);
 
+  // check if token is expired
+  useEffect(() => {
+    const error = query.error as AxiosError;
+    if (error && error.status === 498) {
+      if (noAuthRoutes.includes(window.location.pathname)) return;
+      window.location.replace(Web.Login);
+      cache.remove(CacheKey.AuthToken);
+      cache.remove(CacheKey.User);
```

Comment:
This not the best way to do it. Use `user.logout` and it will handle everyting for you.